generator/internal/dart/dart.go

```diff
@@ -0,0 +1,309 @@
+		Name:    "Duration",
+		Package: "time",
+	}
+	s.MessageByID[timestamp.ID] = timestamp
+	s.MessageByID[duration.ID] = duration
+}
+
+func fieldType(f *api.Field, state *api.APIState, importMap map[string]*goImport) string {
+	var out string
+	switch f.Typez {
+	case api.STRING_TYPE:
+		out = "string"
+	case api.INT64_TYPE:
+		out = "int64"
+	case api.INT32_TYPE:
+		out = "int32"
+	case api.BOOL_TYPE:
+		out = "bool"
+	case api.BYTES_TYPE:
+		out = "[]byte"
+	case api.MESSAGE_TYPE:
+		m, ok := state.MessageByID[f.TypezID]
+		if !ok {
+			slog.Error("unable to lookup type", "id", f.TypezID)
+			return ""
+		}
+		if m.IsMap {
+			key := fieldType(m.Fields[0], state, importMap)
+			val := fieldType(m.Fields[1], state, importMap)
+			out = "map[" + key + "]" + val
+			break
+		}
+		out = "*" + messageName(m, importMap)
+	case api.ENUM_TYPE:
+		e, ok := state.EnumByID[f.TypezID]
+		if !ok {
+			slog.Error("unable to lookup type", "id", f.TypezID)
+			return ""
+		}
+		out = enumName(e, importMap)
+	default:
+		slog.Error("unhandled fieldType", "type", f.Typez, "id", f.TypezID)
+	}
+	return out
+}
+
+func asQueryParameter(f *api.Field) string {
+	return fmt.Sprintf("req.%s.to_str()", strcase.ToLowerCamel(f.Name))
+}
+
+func templatesProvider() language.TemplateProvider {
+	return func(name string) (string, error) {
+		contents, err := dartTemplates.ReadFile(name)
+		if err != nil {
+			return "", err
+		}
+		return string(contents), nil
+	}
+}
+
+func methodInOutTypeName(id string, s *api.APIState) string {
+	if id == "" {
+		return ""
+	}
+	m, ok := s.MessageByID[id]
+	if !ok {
+		slog.Error("unable to lookup type", "id", id)
+		return ""
+	}
+	return strcase.ToCamel(m.Name)
+}
+
+func messageName(m *api.Message, importMap map[string]*goImport) string {
+	if m.Parent != nil {
+		return messageName(m.Parent, importMap) + "_" + strcase.ToCamel(m.Name)
+	}
+	if imp, ok := importMap[m.Package]; ok {
+		return imp.name + "." + toPascal(m.Name)
+	}
+	return toPascal(m.Name)
+}
+
+func enumName(e *api.Enum, importMap map[string]*goImport) string {
+	if e.Parent != nil {
+		return messageName(e.Parent, importMap) + "_" + strcase.ToCamel(e.Name)
+	}
+	return strcase.ToCamel(e.Name)
+}
+
+func enumValueName(e *api.EnumValue, importMap map[string]*goImport) string {
+	var name string
+	if strings.ToUpper(e.Name) == e.Name {
+		name = e.Name
+	} else {
+		name = strcase.ToScreamingSnake(e.Name)
+	}
+	if e.Parent.Parent != nil {
+		return messageName(e.Parent.Parent, importMap) + "_" + name
+	}
+	return name
+}
+
+func bodyAccessor(m *api.Method) string {
+	if m.PathInfo.BodyFieldPath == "*" {
+		// no accessor needed, use the whole request
+		return ""
+	}
+	return "." + strcase.ToCamel(m.PathInfo.BodyFieldPath)
+}
+
+func httpPathFmt(m *api.PathInfo) string {
+	fmt := ""
+	for _, segment := range m.PathTemplate {
+		if segment.Literal != nil {
+			fmt = fmt + "/" + *segment.Literal
+		} else if segment.FieldPath != nil {
+			fmt = fmt + "/%s"
```

Comment:
I suspect this code will need to change for Dart too.  It may be better to just return an empty string until we know what goes here?